flight/util/json_parsing.py

```diff
@@ -0,0 +1,65 @@
+"""Separates data from SUAS mission plan into data structures for use in specific files"""
+from typing import Dict, List
+import json
+
+
+def waypoint_parsing(filename: str) -> List[Dict[str, float]]:
+    """
+    Accepts name of JSON file and extracts waypoint data for SUAS mission
+
+    Parameters
+    ----------
+        filename: str
+            String of data file to open and access waypoint data
+
+    Returns
+    -------
+        List[Dict[str, float]]
+            List of dictionaries containing latitude, longitude and altitude of each waypoint in mission
+
+    Raises
+    ------
+        General
+            Designed to detect any error to prevent data corruption and always close the file being read
+    """
+    with open(filename) as f:
+        try:
+            data_set: Dict[str, List] = json.load(f)
+        except:
+            f.close()
+    f.close()
```

Comment:
You don't need an f.close since you're using a context manager.
Also everything should be either structurally subtyped or `Any`, so you can either replace `List` with `List[Any]` if you're sure that the json is to be loaded in as a dictionary mapping strings to lists or you can replace `List` with just `Any` so that you can account for any type being mapped. Json typing is really not in a good place right now in python, so that's the only reasonably readable way to do it.
The result should look like this:
```python
with open(filename) as f:
    data_set: Dict[str, Any] = json.load(f)
```